SCClassLibrary/Common/Collections/SequenceableCollection.sc

```diff
@@ -842,31 +842,31 @@ SequenceableCollection : Collection {
 	lgamma { ^this.performUnaryOp('lgamma') }
 	digamma { ^this.performUnaryOp('digamma') }
 	trigamma { ^this.performUnaryOp('trigamma') }
-	polygamma { |n, z| ^this.performBinaryOp('polygamma', n, z) }
-	tgammaRatio { |a, b| ^this.performBinaryOp('tgammaRatio', a, b) }
-	tgammaDeltaRatio { |a, delta| ^this.performBinaryOp('tgammaDeltaRatio', a, delta) }
-	gammaP { |a, z| ^this.performBinaryOp('gammaP', a, z) }
-	gammaQ { |a, z| ^this.performBinaryOp('gammaQ', a, z) }
-	tgammaLower { |a, z| ^this.performBinaryOp('tgammaLower', a, z) }
-	tgammaUpper { |a, z| ^this.performBinaryOp('tgammaUpper', a, z) }
+	polygamma { |n, z| ^this.multiChannelPerform('polygamma', n, z) }
```

Comment:
Is this a semantic distinction? I had (naively) assumed that `performBinaryOp` would be used for any of the functions with 2 arguments, and the simple tests  I did showed correct behavior:
```
polygamma(1,4);  -> 0.28382295573712
polygamma(2,5);  -> -0.048789732245114
polygamma(3,6);  -> 0.011827828192755

polygamma([1,2,3],[4,5,6])
-> [ 0.28382295573712, -0.048789732245114, 0.011827828192755 ]
```
After doing some research, the distinction still isn't clear to me. Would you mind briefly describing the difference?

Reply:
these messages are not binary ops, the third argument of a binary op is always an *adverb*. For example, this succeeds:

```
polygamma(1, 2, 3)
```

while this fails:
```
polygamma([1,2,3],[4,5,6],[1,2,3]) // ERROR: unrecognized adverb: '[ 1, 2, 3 ]' for operator 'polygamma'
```

For expanding n-ary ops, we therefore have to use `multiChannelPerform`.

Reply:
> I had (naively) assumed that `performBinaryOp` would be used for any of the functions with 2 arguments

These are all operators with *three* arguments, even though they are methods with two: if you switch perspective, this is what happens. For example, the binary operator `+` is a method with only one argument, because the receiver is the first operand.

This is mirrored in the two ways in which you can write down such expressions:

`3.polygamma(4, 2)`
or
`polygamma(3, 4, 2)`
depending on what makes more sense in the situation

Reply:
P.S. The `adverb` is a special thing altogether, not meant to extend the number of arguments of the operator, but a way to specify how it is applied.

Reply:
Thanks for the clarification!